gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -189,6 +197,10 @@ public AbstractJobLauncher(Properties jobProps, List<? extends Tag<?>> metadataT
     clusterNameTags.addAll(Tag.fromMap(ClusterNameTags.getClusterNameTags()));
     GobblinMetrics.addCustomTagsToProperties(jobProps, clusterNameTags);
 
+    issueRepository = new InMemoryIssueRepository();
+    troubleshooter = new AutomaticTroubleshooter(issueRepository);
```

Comment:
Instead of enabling this in general for all gobblin jobs, should we consider making this troubleshooter opt-in to start with, or at least have a config to disable it? Just in case there are unexpected performance issues, it floods logs too much, it's sending too many events, etc.

Reply:
Added configs to disable event reporting and all troubleshooter activity in case of unexpected problems